service/src/test/java/eu/europa/ec/fisheries/uvms/user/service/arquillian/BuildUserServiceDeployment.java

```diff
@@ -0,0 +1,59 @@
+package eu.europa.ec.fisheries.uvms.user.service.arquillian;
+
+import org.eu.ingwar.tools.arquillian.extension.suite.annotations.ArquillianSuiteDeployment;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.resolver.api.maven.Maven;
+import org.junit.Assert;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+@ArquillianSuiteDeployment
+public abstract class BuildUserServiceDeployment extends Assert {
+
+    final static Logger LOG = LoggerFactory.getLogger(BuildUserServiceDeployment.class);
+
+
+    @Deployment(name = "normal", order = 1)
+    public static Archive<?> createDeployment() {
+
+        // Embedding war package which contains the test class is needed
+        // So that Arquillian can invoke test class through its servlet test runner
+        WebArchive testWar = ShrinkWrap.create(WebArchive.class, "test.war");
+        testWar.addPackages(true,"com.tocea.easycoverage.framework.api");
+        testWar.addPackages(true,"eu.europa.ec.fisheries.uvms.user.service");
+        testWar.addPackages(true,"eu.europa.ec.fisheries.uvms.user.service.bean");
+        testWar.addPackages(true,"eu.europa.ec.fisheries.uvms.user.service.dao");
+        testWar.addPackages(true,"eu.europa.ec.fisheries.uvms.user.service.dto");
+        testWar.addPackages(true,"eu.europa.ec.fisheries.uvms.user.service.entity");
+        testWar.addPackages(true,"eu.europa.ec.fisheries.uvms.user.service.config");
+        testWar.addPackages(true,"eu.europa.ec.fisheries.uvms.user.service.converter");
+        testWar.addPackages(true,"eu.europa.ec.fisheries.uvms.user.service.exception");
+        testWar.addPackages(true,"eu.europa.ec.mare.usm.information");
+        testWar.addPackages(true,"eu.europa.ec.mare.usm.administration");
+        testWar.addPackages(true,"eu.europa.ec.fisheries.wsdl.user");
+        testWar.addPackages(true,"eu.europa.ec.fisheries.uvms");
+        testWar.addPackages(true,"eu.europa.ec.mare.usm.policy.service");
+        testWar.addPackages(true,"eu.europa.ec.mare.usm.service");
+        testWar.addPackages(true,"eu.europa.ec.mare.usm.authentication");
+        testWar.addPackages(true,"eu.europa.ec.mare.usm.session");
+        testWar.addPackages(true,"eu.europa.ec.mare.usm.session");
+        //find a way to reference USM PU otherwise it fails
+        testWar.addClass(TransactionalTests.class);
```

Comment:
Referencing the subclass from the superclass is somewhat scary, we have to take a careful look, when and if we activate Arquillian.